python/lsst/daf/butler/_butler.py

```diff
@@ -1131,6 +1131,43 @@ def getURI(self, datasetRefOrType: Union[DatasetRef, DatasetType, str],
                                "Use Butler.getURIs() instead.")
         return primary
 
+    def getArtifacts(self, refs: Iterable[DatasetRef],
+                     destination: Union[str, ButlerURI], transfer: str = "auto",
+                     preserve_path: Optional[bool] = True) -> List[ButlerURI]:
+        """Retrieve the artifacts associated with the supplied refs.
+
+        Parameters
+        ----------
+        refs : iterable of `DatasetRef`
+            The datasets for which artifacts are to be retrieved.
+            A single ref can result in multiple artifacts. The refs must
+            be resolved.
+        destination : `ButlerURI` or `str`
+            Location to write the artifacts.
+        transfer : `str`, optional
+            Method to use to transfer the artifacts. Must be one of the options
+            supported by `ButlerURI.transfer_from()`.
+        preserve_path : `bool`, optional
+            If `True` the full path of the artifact within the datastore
+            is preserved. If `False` the final file component of the path
```

Comment:
What happens if the final file components collide, or should that never happen?

Reply:
The transfer will fail and everything will stop. It shouldn't happen because we are meant to check that the file itself is unique based on the associated dimensions and collection.

Reply:
That also means that an interrupted retrieve cannot be retried in place without cleaning up first.  Maybe that's obvious, but it should be documented.

Reply:
I can add a `clobber` option that will always overwrite. In theory I can make it transactional and have all the transfers roll back on error.

Reply:
Added a clobber/overwrite option. Could also check whether the file exists and skip the transfer if it does. Can leave that to a follow up ticket if that is wanted.